### PR TITLE
Fixes l10n workflow.

### PR DESCRIPTION
**Pull request by pierreozoux**
I think this time it will be good \o/

(sorry for all these commits, difficult to try on my machine :/ )

- I add the id to the job so the test to autoapprove/automerge will work
- I avoid to test this branch as this branch is triggered by a bot and then [it doens't trigger other jobs](https://stackoverflow.com/a/69169858)...

### CODE REVIEW

File: .github/workflows/sphinxbuild.yml
Comment (pierreozoux) [reply .github/workflows/sphinxbuild.yml:9]: This action does only create the [translation files picked up by transifex](https://github.com/nextcloud/documentation/blob/master/.github/workflows/generate_catalog_templates.yml#L19=) so it shouldn't touch anything related to the build.

And we need to do that to have the automerge feature.

basically we have 2 choices:
 - either put an admin token to be able to automerge
 - do what I do in this PR (avoid having checks on the PR, to be able to automerge)

It has been months I'm on this topic now, and I don't see much more options :)

Comment (pierreozoux) [reply .github/workflows/sphinxbuild.yml:9]: @nickvergessen can you accept it and we merge? We can have a quick call also that I can explain :)

**Comment by skjnldsv** [comment]
> * I avoid to test this branch as this branch is triggered by a bot and then [it doens't trigger other jobs](https://stackoverflow.com/a/69169858)...

You can use appropriate tokens for those.
Have a look at the transifex configuration we have here for example
https://github.com/nextcloud/nextcloud-upload/blob/master/.github/workflows/l10n.yml

**Comment by pierreozoux** [comment]
@skjnldsv Actually, let me go back and take a bit of perspective.

The transifex -> GitHub problem is [already solved](https://github.com/nextcloud/documentation/blob/master/.github/workflows/transifex.yml). (Transifex automatically creates a PR and it is automagically merged \o/)

But, the missing bit is how to do github -> transifex automatically.

To my knowledge here are the approaches possible:
 - ask people contributing to doc to also include updated pot files (I think it is a lot of work on their side, and will not ease contribution)
 - run a job against PR to add this change (I don't like this approach as I find it difficult to amend work in PR)
 - run a job once a PR is merged
   - this job could automatically commit to master branch the pot files (this is what I'm trying to do in this PR) (and then transifex is configured to pull from github)
   - or this job could push directly to transifex with the transifex cli

I think the last approach is my favorite one. I like to have one version of resources, and not copy them around.
So for me the ideal case would be to:
 - once a PR with new doc is merged
 - create POT files in a job
 - push these to transifex
 - and do not store them in GitHub
And then, once we want to build documentation:
 - pull PO files (translated strings) from transifex
 - build doc with these pulled files

The benefit is that we avoid the noise of PRs and actions moving around all the time.

But to do so, we need:
 - transifex api token (from me or a bot, don't know the best)
 -  access to GitHub settings to configure this token in the settings.
 
 As I don't have either, that's why I went the "automatic PR" route.
 
 And then once we are happy with the automation, we can replicate it in the various Nextcloud repos.
 
 Hope this helps, and I'd love to have your feedback @skjnldsv  :) 
 
 Have a nice day!

**Comment by skjnldsv** [comment]
Here is the config for the upload library
image

**Comment by pierreozoux** [comment]
@skjnldsv yes ok, I know, I configured it in this repo :upside_down_face: 

But how to you convert rst files to pot files automatically without asking "devs" to do so? This is what I try to fix.

**Comment by skjnldsv** [comment]
> But how to you convert rst files to pot files automatically without asking "devs" to do so? This is what I try to fix.

Ah, sorry, I misunderstood this part: 
> But, the missing bit is how to do github -> transifex automatically.

I would ask the devs, same as we do in a lot of places.
You can also add a command to do so from a comment, [same as we do for the shipped binaries](https://github.com/nextcloud/.github/blob/master/workflow-templates/command-compile.yml). But instead you use the l10n pot generate command

**Comment by pierreozoux** [comment]
> You can also add a command to do so from a comment, [same as we do for the shipped binaries](https://github.com/nextcloud/.github/blob/master/workflow-templates/command-compile.yml). But instead you use the l10n pot generate command
> 
> `/l10n-build (amend|fixup)?` or something

Then this is similar to
> run a job against PR to add this change (I don't like this approach as I find it difficult to amend work in PR)

I wouldn't say that this approach is fully automatic either.

I'm also available for a call, maybe it is easier, and I think it is worth it to discuss a bit, and have the same approach on all repos.

**Comment by pierreozoux** [comment]
I thought a lot about this topic, and I think the right approach would be:

Push to transifex:
```
- once a PR with new doc is merged
- create POT files in a job
- push these to transifex (and do not store them in GitHub)
```

Pull from transifex, once we want to build documentation:
```
 - pull PO files (translated strings) from transifex
 - build doc with these pulled files
```

If you agree with this approach, I can start to implement it.

If you have questions, suggestions, want to discuss what would be the best approach, let's sit down around the table with a 30m/1h call together to fix doc and translation for Nc once for all.

**Comment by p-bo** [comment]
[skjnldsv](https://github.com/skjnldsv) With all respect to aim to ensure the most correct workflow, in meantime, we have translated user docs for servers into quite few languages for long time, even with imperfect workflow, thus much more useful for user around the globe. But unfortunately translations of desktop client docs are blocked until this will be solved in most perfect way. So what about to unblock it allowing the same workflow as for user-server docs for now (so we, translators could do work for user) and afterward refactor both workflows "on background" into desired state?

**Comment by p-bo** [comment]
Any hope we can advance there please? That offered virtual meeting between repo admins and contributors sounded promising…

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
Let's move forward

**Pull request merged**
